godot-core/src/builtin/rect2i.rs

```diff
@@ -78,21 +82,492 @@ impl Rect2i {
         self.size = end - self.position
     }
 
-    /* Add in when `Rect2i::abs()` is implemented.
+    /// Returns a `Rect2i` with equivalent position and area, modified so that the top-left corner
+    /// is the origin and `width` and `height` are positive.
+    #[inline]
+    pub fn abs(self) -> Self {
+        let abs_size = self.size.abs();
+        let offset = Vector2i::new(cmp::min(self.size.x, 0), cmp::min(self.size.y, 0));
+        Self::new(self.position + offset, abs_size)
+    }
+
+    /// Returns `true` if this `Rect2i` completely encloses another one.
+    #[inline]
+    pub const fn encloses(&self, other: Self) -> bool {
+        self.assert_nonnegative();
+        other.assert_nonnegative();
+        // FIXME: provide an unsafe version without the check?
+        // FIXME: or just remove the check altogether?
+        let own_end = self.end();
+        let other_end = other.end();
+        other.position.x >= self.position.x
+            && other.position.y >= self.position.y
+            && other_end.x <= own_end.x
+            && other_end.y <= own_end.y
+    }
+
+    /// Returns a copy of this `Rect2i` expanded so that the borders align with the given point.
+    #[inline]
+    pub fn expand(self, to: Vector2i) -> Self {
+        self.assert_nonnegative();
+        // FIXME: provide an unsafe version without the check?
+        // FIXME: or just remove the check altogether?
+        let mut end = self.end();
+        let mut begin = self.position;
+
+        begin.x = cmp::min(begin.x, to.x);
+        begin.y = cmp::min(begin.y, to.y);
+        end.x = cmp::max(end.x, to.x);
+        end.y = cmp::max(end.y, to.y);
+
+        Self::from_corners(begin, end)
+    }
+
+    /// Returns the area of the `Rect2i`.
+    #[inline]
+    pub const fn get_area(&self) -> i32 {
+        self.size.x * self.size.y
+    }
+
+    /// Returns the center of the `Rect2i`, which is equal to `position + (size / 2)`.
+    ///
+    /// If `size` is an odd number, the returned center value will be rounded towards `position`.
+    #[inline]
+    pub fn get_center(&self) -> Vector2i {
+        self.position + (self.size / 2)
+    }
+
+    /// Returns a copy of the `Rect2i` grown by the specified `amount` on all sides.
+    #[inline]
+    pub fn grow(self, amount: i32) -> Self {
+        let amount_2d = Vector2i::new(amount, amount);
+        Self::from_corners(self.position - amount_2d, self.end() + amount_2d)
+    }
+
+    /// Returns a copy of the `Rect2i` grown by the specified amount on each side individually.
+    #[inline]
+    pub const fn grow_individual(self, left: i32, top: i32, right: i32, bottom: i32) -> Self {
+        Self::from_components(
+            self.position.x - left,
+            self.position.y - top,
+            self.size.x + left + right,
+            self.size.y + top + bottom,
+        )
```

Comment:
It would be simpler and more symmetric to reuse `Self::from_corners()` here, no?